keyboards/aki27/cocot46plus/cocot46plus.c

```diff
@@ -0,0 +1,280 @@
+            cocot_config.scrl_mode ^= 1;
+        }
+    }
+
+    return true;
+}
+
+void eeconfig_init_kb(void) {
+    cocot_config.cpi_idx = COCOT_CPI_DEFAULT;
+    cocot_config.scrl_div = COCOT_SCROLL_DIV_DEFAULT;
+    cocot_config.rotation_angle = COCOT_ROTATION_DEFAULT;
+    cocot_config.scrl_inv = COCOT_SCROLL_INV_DEFAULT;
+    cocot_config.scrl_mode = false;
+    eeconfig_update_kb(cocot_config.raw);
+    eeconfig_init_user();
+    adns5050_write_reg(0x22, 0b10000 | 0x80);
+}
+
+
+void matrix_init_kb(void) {
+    // is safe to just read CPI setting since matrix init
+    // comes before pointing device init.
+    cocot_config.raw = eeconfig_read_kb();
+    if (cocot_config.cpi_idx > CPI_OPTION_SIZE) // || cocot_config.scrl_div > SCRL_DIV_SIZE || cocot_config.rotation_angle > ANGLE_SIZE)
+    {
+        eeconfig_init_kb();
+    }
+    matrix_init_user();
+}
+
+
+bool cocot_get_scroll_mode(void) {
+    return cocot_config.scrl_mode;
+}
+
+void cocot_set_scroll_mode(bool mode) {
+    cocot_config.scrl_mode = mode;
+}
+
+
+
+// OLED utility
+#ifdef OLED_ENABLE
+
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+    return OLED_ROTATION_0;
+}
+
+static const char PROGMEM cocot_logo[] = {
+    0x80, 0x81, 0x82, 0x83, 0x84, 0x85, 0x86, 0x87, 0x88, 0x89, 0x8A, 0x8B, 0x8C, 0x8D, 0x8E, 0x8F, 0x90, 0x91, 0x92, 0x93, 0x94,
+    0xA0, 0xA1, 0xA2, 0xA3, 0xA4, 0xA5, 0xA6, 0xA7, 0xA8, 0xA9, 0xAA, 0xAB, 0xAC, 0xAD, 0xAE, 0xAF, 0xB0, 0xB1, 0xB2, 0xB3, 0xB4,
+    0xC0, 0xC1, 0xC2, 0xC3, 0xC4, 0xC5, 0xC6, 0xC7, 0xC8, 0xC9, 0xCA, 0xCB, 0xCC, 0xCD, 0xCE, 0xCF, 0xD0, 0xD1, 0xD2, 0xD3, 0xD4,
+    0};
+
+void render_logo(void) {
+    oled_write_P(cocot_logo, false);
+};
+
+void oled_write_layer_state(void) {
+
+    oled_write_P(PSTR(" "), false);
+
+    switch (get_highest_layer(layer_state | default_layer_state)) {
+        case 0:
+            oled_write_P(PSTR("Base "), false);
+            break;
+        case 1:
+            oled_write_P(PSTR("Lower"), false);
+            break;
+        case 2:
+            oled_write_P(PSTR("Raise"), false);
+            break;
+        case 3:
+            oled_write_P(PSTR("Mouse"), false);
+            break;
+        case 4:
+            oled_write_P(PSTR("L4   "), false);
+            break;
+        case 5:
+            oled_write_P(PSTR("L5   "), false);
+            break;
+        case 6:
+            oled_write_P(PSTR("L6   "), false);
+            break;
+        default:
+            oled_write_P(PSTR("Undef"), false);
+            break;
+    }
+    oled_write_P(PSTR("/"), false);
+    if (cocot_get_scroll_mode()){
+        oled_write_P(PSTR("S"), false);
+    } else{
+        oled_write_P(PSTR("C"), false);
+    }
+
+    char cpi[5];
+    char scroll_div[3];
+    char angle[4];
+    snprintf(cpi, 5, "%4d", cpi_array[cocot_config.cpi_idx]);
+    snprintf(scroll_div, 3, "%2d", scrl_div_array[cocot_config.scrl_div]);
+    snprintf(angle, 4, "%3d",  angle_array[cocot_config.rotation_angle]);
```

Comment:
just a heads up, you can use `get_u8_str(x, char)` and `get_u16_str(x, char)` to generate a 3 and 5 character array with the number values.

Reply:
Thanks for the tip! I'm going to pass in this case, but I'll keep it in the mind for the future.